<a href="https://colab.research.google.com/github/kgd1020/varios_objetos/blob/main/Estagio_Sprint_8_Tarefa_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### ESTÁGIO SPRINT-8 TAREFA-3

### [Warm up] Em Python, declare e inicialize uma lista contendo 250 inteiros obtidos de forma aleatória. Após, aplicar o método reverse sobre o conteúdo da lista e imprimir o resultado.

In [2]:
import random

In [3]:
lista_inteiros = [random.randint(0, 1000) for _ in range(250)]

In [4]:
lista_inteiros.reverse()
print(lista_inteiros)

[693, 166, 415, 337, 70, 557, 878, 502, 26, 404, 756, 525, 1000, 798, 340, 438, 710, 734, 847, 174, 635, 175, 969, 996, 453, 333, 711, 537, 325, 124, 933, 200, 389, 188, 315, 763, 275, 266, 10, 925, 190, 135, 762, 856, 963, 697, 48, 182, 309, 383, 774, 358, 427, 989, 265, 450, 533, 701, 653, 23, 895, 605, 875, 215, 806, 250, 575, 648, 999, 907, 871, 856, 232, 845, 889, 171, 180, 767, 941, 662, 897, 922, 942, 702, 918, 731, 126, 646, 793, 612, 237, 757, 764, 70, 735, 924, 917, 497, 412, 732, 446, 360, 791, 410, 459, 215, 604, 757, 247, 551, 721, 339, 719, 539, 382, 202, 985, 13, 977, 685, 385, 519, 433, 31, 598, 749, 162, 157, 738, 975, 361, 729, 493, 599, 655, 979, 771, 299, 469, 776, 466, 927, 133, 567, 285, 544, 322, 637, 793, 101, 862, 881, 986, 421, 72, 357, 35, 620, 201, 939, 551, 215, 786, 520, 221, 347, 583, 34, 246, 503, 820, 212, 298, 701, 974, 839, 703, 975, 200, 497, 446, 328, 93, 908, 773, 772, 47, 664, 928, 27, 665, 574, 558, 68, 118, 758, 662, 766, 522, 78, 932, 426, 670,

### [Warm up] Em Python, declare e inicialize uma lista contendo o nome de 20 animais. Ordene-os em ordem crescente e itere sobre os itens, imprimindo um a um (você pode utilizar list comprehension aqui).  Na sequência, armazene o conteúdo da lista em um arquivo de texto, um item em cada linha, no formato CSV.

In [5]:
animais = ["cachorro", "gato", "elefante", "leão", "girafa", "macaco", "coelho", "cavalo", "tigre", "pato",
           "papagaio", "canguru", "peixe", "formiga", "abelha", "zebra", "coruja", "urso", "golfinho", "camelo"]

In [6]:
animais_ordenados = sorted(animais)
print([animal for animal in animais_ordenados])

['abelha', 'cachorro', 'camelo', 'canguru', 'cavalo', 'coelho', 'coruja', 'elefante', 'formiga', 'gato', 'girafa', 'golfinho', 'leão', 'macaco', 'papagaio', 'pato', 'peixe', 'tigre', 'urso', 'zebra']


In [8]:
with open("animais.csv", "w", encoding="utf-8") as arquivo:
    for animal in animais_ordenados:
        arquivo.write(f"{animal}\n")

### [Laboratório] Elaborar um código Python para gerar um dataset de nomes de pessoas. Siga os passos a seguir para realizar a atividade:

### Passo 1:  Instalar biblioteca names para geração de nomes aleatórios. O comando de instalação é pip install names

In [9]:
pip install names

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 789.1/789.1 kB 45.3 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Created wheel for names: filename=names-0.3.0-py3-none-any.whl size=803682 sha256=197f6ba0b3fc40e74b65e49a9826a8b9b833b52f0976b032f79e58fde57475cf
  Stored in directory: /root/.cache/pip/wheels/fc/9a/6f/78f4282bbcaa2d8c678b73c54c0bb1b7a04009f0d7cec79fce
Successfully built names


### Passo 2: Importar as bibliotecas random, time, os e names em seu código

In [10]:
import random
import time
import os
import names

### Passo 3: Definir os parâmetros para geração do dataset, ou seja, a quantidades de nomes aleatórios e a quantidade de nomes que devem ser únicos.


### Define a semente de aleatoriedade

In [11]:
random.seed(40)

qtd_nomes_unicos = 3000
qtd_nomes_aleatorios = 10000000

### Gerar os nomes aleatórios e um arquivo de texto com esses nomes

In [14]:
# Gerar os nomes aleatórios
lista_nomes_unicos = []
for i in range(0, qtd_nomes_unicos):
    lista_nomes_unicos.append(names.get_full_name())

print("Gerando {} nomes aleatórios".format(qtd_nomes_aleatorios))
dados = []

for i in range(0, qtd_nomes_aleatorios):
    dados.append(random.choice(lista_nomes_unicos))

# Gerar um arquivo de texto com os nomes
nome_arquivo = "nomes_aleatorios.txt"
with open(nome_arquivo, "w") as arquivo:
    for nome in dados:
        arquivo.write(nome + "\n")
os.system(nome_arquivo)

Gerando 10000000 nomes aleatórios


32512

### Abrir o arquivo e verificar seu conteúdo. O valor de retorno "32512" indica que o comando foi executado com sucesso, mas não houve saída disponível para ser exibida. Isso acontece quando o comando não produz saída para ser exibida no terminal.

In [15]:
retorno = os.system(nome_arquivo)
print(retorno)


32512
